src/software/ai/evaluation/detect_successful_kick.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include "software/geom/vector.h"
+#include "software/world/ball.h"
+
+/**
+ * Software approximation that finds if a ball has been kicked, regardless of whether the
+ * kick was a pass, shot, or chip.
+ *
+ * @param ball The ball that can be kicked
+ * @param kick_direction The direction that we expect the ball to be kicked towards
+ *
+ * @return True if ball was kicked in the approximate direction we expect, false otherwise
+ */
+bool successfulKickDetected(const Ball& ball, const Vector& kick_direction);
```

Comment:
I'm wondering if we should have this in "software/world/world.h" or even "software/world/ball.h". Having it in evaluation requires us to pass in the ball as a parameter when it is effectively checking the ball's state. It seems a little weird to me, thoughts? @jonathanlew @akhilveeraghanta @kelvinkoon @amrm3lm

Reply:
I think thats a fair point. Though I'd say its doing more than checking the ball's state, since it includes `kick_direction` which is entirely dependent on the play/action/robot.

Reply:
I like the idea of putting it in `Ball` @akhilveeraghanta thoughts?

Reply:
Detecting if the ball has been kicked in a certain direction is an operation on the balls current state rather than a property of the ball imo. We have always treated the `Ball` class as a "data structure" which just holds onto the data, and any operation we would perform on a `Ball` instance, would take a reference to the `Ball` and operate on that.

So for that reason I think having it as an evaluation function makes the most sense. This function takes a const reference to the ball so _shouldn't_ be too bad.

@kelvinkoon @jonathanlew @amrm3lm @jared-paul thoughts?

Reply:
Well Ball also has BallModel and history so it's not a simple data structure. What benefit do we get from having Ball being more simple?

Reply:
There's files like `detect_thread.cpp` where `calcBallVelIntersectFriendlyNet` takes in `Ball` as well, so I think it if we follow that, being in `evaluation` makes sense. That being said, I'm not against it being in `Ball`.

Reply:
> Well Ball also has BallModel and history so it's not a simple data structure. What benefit do we get from having Ball being more simple?

Yeah def no benefit in keeping the class simple, what I mean't was more like everything in the ball is a property of the ball, the `BallModel` is entirely owned by the ball.

For functions like `ballInEnemyHalf`, `ballInFriendlyHalf` and whatnot, we pass in a `Ball` and a `Field` and the evaluation function interprets position. 

So if we made this change, does it raise the question if ball should do the following as well?

* `ball.isInEnemyHalf(Field f)` 
* `ball.isInFriendlyHalf(Field f)`
* `ball.isInFriendlyCorner(Field f)`
* `ball.isInEnemyCorner(Field f)`

We would have to pass a field in because a ball doesn't own a field, similar to how a ball doesn't own the direction its being kicked. 

* `ball.hasBeenKicked(30)`

Thats sort of where I am coming from, Im not _entirely_ against moving it to the ball, but it's worth considering where this function stands compared to the current Ball evaluation functions we have.

Reply:
Well arguably `ballInEnemyHalf` and friends should be replaced by `contains(field.enemyHalf(), ball.position())`. 
I don't feel strongly about it being in the `Ball` class, but I suppose it should go in `evaluation/ball.h` to be consistent.

Reply:
So i guess consensus is that its not necessary for it to be in the `ball` class. The suggestion that it be in `evaluation/ball.h` makes sense , and since it is not being opposed, I'll move the method there @akhilveeraghanta @jonathanlew